api/controllers/UserController.js

```diff
@@ -1071,13 +1071,27 @@ module.exports = {
    *   '404':
    *     description: Requested user not found.
    */
-  async updatePassword(request, reply) {
-    const user = await User.findOne({ _id: request.params.id });
+  async updatePassword(request, reply, internalArgs) {
+    let userId, old_password, new_password, confirm_password;
+
+    if (internalArgs) {
+      userId = internalArgs.userId;
+      old_password = internalArgs.old_password;
+      new_password = internalArgs.new_password;
+      confirm_password = internalArgs.confirm_password;
```

Comment:
Good eye, it isn't yet. IMO it should and so eventually we'll be using the variable.

As it stands on prod and in the old app, you could bypass client-side validation and accidentally send a password you didn't confirm correctly, and it will happily use the `new_password` as the value.